server/test/integration/com/thoughtworks/go/server/service/UserServiceIntegrationTest.java

```diff
@@ -609,6 +611,73 @@ public void shouldDisableAdminSelectionWhenUserIsMemberOfAdminRole() throws Exce
         assertThat(userService.getAdminAndRoleSelections(Arrays.asList("foo", "quux")).getAdminSelection(), is(new TriStateSelection(Admin.GO_SYSTEM_ADMIN, TriStateSelection.Action.add, false)));
     }
 
+    @Test
+    public void updateShouldUpdateEnabledState() throws Exception {
+        userDao.saveOrUpdate(new User("admin-user"));
+        userDao.saveOrUpdate(new User("user-1"));
+        userService.enable(Arrays.asList("user-1"), new HttpLocalizedOperationResult());
+
+        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.FALSE, TriState.UNSET, null, null, result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").isEnabled(), is(false));
+
+        result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.TRUE, TriState.UNSET, null, null, result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").isEnabled(), is(true));
+    }
+
+    @Test
+    public void updateShouldUpdateEmail() throws Exception {
+        userDao.saveOrUpdate(new User("user-1"));
+
+        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.UNSET, TriState.UNSET, "foo@example.com", null, result);
+        assertThat(result.isSuccessful(), is(true));
+
+        assertThat(userService.findUserByName("user-1").getEmail(), is("foo@example.com"));
+
+        result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.TRUE, TriState.UNSET, "", null, result);
+        assertThat(userService.findUserByName("user-1").getEmail(), is(""));
+        assertThat(result.isSuccessful(), is(true));
+    }
+    @Test
+    public void updateShouldUpdateMatcher() throws Exception {
+        userDao.saveOrUpdate(new User("user-1"));
+
+        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.UNSET, TriState.UNSET, null, "foo,bar", result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").getMatcher(), is("foo,bar"));
+
+        result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.UNSET, TriState.UNSET, null, "", result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").getMatcher(), is(""));
+    }
+
+    @Test
+    public void updateShouldUpdateEmailMeState() throws Exception {
+        userDao.saveOrUpdate(new User("user-1"));
+
+        User user = userService.findUserByName("user-1");
+        user.setEmailMe(false);
+        userDao.saveOrUpdate(user);
+
+        userService.enable(Arrays.asList("user-1"), new HttpLocalizedOperationResult());
```

Comment:
I know,
 But I found it tough to read one method doing a lot of operations.

 I feel it's cleaner to have one method call and one assertion for the
result of that method call. So that I can easily map what the assertion is
testing.

On Thu, Jul 2, 2015 at 1:33 PM, Ketan Padegaonkar <notifications@github.com>
wrote:

> In
> server/test/integration/com/thoughtworks/go/server/service/UserServiceIntegrationTest.java
> <https://github.com/gocd/gocd/pull/1261#discussion_r33754873>:
>
> > +
> > +        result = new HttpLocalizedOperationResult();
> > +        userService.update(new User("user-1"), TriState.UNSET, TriState.UNSET, null, "", result);
> > +        assertThat(result.isSuccessful(), is(true));
> > +        assertThat(userService.findUserByName("user-1").getMatcher(), is(""));
> > +    }
> > +
> > +    @Test
> > +    public void updateShouldUpdateEmailMeState() throws Exception {
> > +        userDao.saveOrUpdate(new User("user-1"));
> > +
> > +        User user = userService.findUserByName("user-1");
> > +        user.setEmailMe(false);
> > +        userDao.saveOrUpdate(user);
> > +
> > +        userService.enable(Arrays.asList("user-1"), new HttpLocalizedOperationResult());
>
> Technically it's a tristate, we'd need 3 different tests for each of the
> tristates, plus 2 additional tests for EACH for email and checkinAliases
> (with null values and without null) I feel these tests are being too
> verbose, and perhaps don't as much value as I'd thought.
>
> WDYT?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gocd/gocd/pull/1261/files#r33754873>.
>